src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -90,9 +96,13 @@ class TcgStatics {
   static flip (String info, Closure eachHead, Closure eachTail={}, multi=[:]){
     flip(info, 1, eachHead, eachTail, multi)
   }
-  static flip (String info, int count, Closure eachHead, Closure eachTail={}, multi=[:]){
+  static flip (String info, int count, Closure eachHead, Closure eachTail={}, multi=[:], playerType=null){
+    if(playerType == null) {
+      playerType = bg.getCurrentTurn()
+    }
     CoinFlip cf=new CoinFlip(count, toEffect (eachHead), toEffect (eachTail))
     cf.setInfo(info)
+    cf.setPlayer(playerType)
```

Comment:
```suggestion
    CoinFlip cf=new CoinFlip(count, toEffect (eachHead), toEffect (eachTail))
    cf.setInfo(info)
    if(playerType) {
      cf.setPlayer(playerType)
    }
```

Reply:
The resolution decision should be left to `CoinFlip` instead of here.

Reply:
Hmm, testing this, the issue I'm having is that player is null `BEFORE` the coin flip, so I can't check the player then, and since AFAICT you can't access the current player without access to the battleground. This works fine otherwise though.

The use case for my issue is Will from CEC. I'm trying to make sure he can only be used by the player playing him instead of the opponent. It works fine how I had it, but gets a null player this way. I can check for this using `if ((ef.player && ef.player == bg.getCurrentTurn()) || !ef.player)` and it seems to work as intended since opponent coin flips explicitly set their player value now, but seems slightly awkward to do. If you don't find it an issue because it still works though, and it probably isn't a very common effect, I can leave it as is.

Reply:
The use case for my issue is Will from CEC. I'm trying to make sure he can only be used by the player playing him instead of the opponent. It works fine how I had it, but gets a null player this way. I can check for this using if ((ef.player && ef.player == bg.getCurrentTurn()) || !ef.player) and it seems to work as intended since opponent coin flips explicitly set their player value now, but seems slightly awkward to do. If you don't find it an issue because it still works though, and it probably isn't a very common effect, I can leave it as is.

Do you mean if Will gets used by another player via an effect like this?

Smeargle (UD 8)
Portrait: Once during your turn (before your attack), if Smeargle is your Active Pokémon, you may look at your opponent's hand. If you do, choose a Supporter card you find there and use the effect of that card as the effect of this power. This power can't be used if Smeargle is affected by a Special Condition.

If yes, Those effects should be wrapped with bg.setDeterministicCurrentThreadPlayerType() call.

Reply:
Not relevant 1-1 but good to think of; 

Another use case to be aware of, is when Portrait is used with Will. Portrait should handle it by wrapping Will's call with `bg.setDeterministicCurrentThreadPlayerType()`. 

> Smeargle (UD 8)
> Portrait: Once during your turn (before your attack), if Smeargle is your Active Pokémon, you may look at your opponent's hand. If you do, choose a Supporter card you find there and use the effect of that card as the effect of this power. This power can't be used if Smeargle is affected by a Special Condition.

Reply:
> Ok I now understand what you mean. That is an interesting use case. There are couple of options;
> 
> 1. resolve player when null in contructor of CoinFlip (but this won't work with legacy sets due to single-instance usage)
> 2. resolve it in `getPlayer()` method (still weird)
> 3. I'd use this condition in Will instead: `(ef.player ?: bg.currentThreadPlayerType) == bg.currentThreadPlayerType`

I've updated this comment multiple times, not sure you've seen my last edit so tagging you @flagrama

Reply:
I'll give that a shot when I get back to this again.